lib/providers/user.dart

```diff
@@ -1,9 +1,77 @@
+import 'dart:convert';
+
 import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import '../util.dart';
+
+enum AttendanceType { In, Out }
 
 class User with ChangeNotifier {
+  String _urlReports = "https://attendance-dcecd.firebaseio.com/reports/";
   String _idToken; // Firebase ID token of the account.
   String _localId; // The uid of the current user.
   String email;
   String displayName;
   String photoUrl;
+
+  Future<dynamic> checkAttendance() async {
+    DateFormat formattedDate = DateFormat('yyyy-MM-dd');
+    String currentDate = formattedDate.format(DateTime.now());
+    String urlCheckAttendance = "$_urlReports$_localId/$currentDate.json?auth=$_idToken";
+    var reponseCheckText = await Util.fetch(FetchType.GET, urlCheckAttendance);
+    final responseData = json.decode(reponseCheckText);
+    return responseData;
```

Comment:
must return attendance status/message from here (function name is checkAttendance)